staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -284,23 +303,65 @@ func (a *GenericAdmissionWebhook) callHook(ctx context.Context, h *v1alpha1.Exte
 }
 
 func (a *GenericAdmissionWebhook) hookClient(h *v1alpha1.ExternalAdmissionHook) (*rest.RESTClient, error) {
+	if client, ok := a.cache.get(&h.ClientConfig); ok {
+		return client, nil
+	}
+
 	serverName := h.ClientConfig.Service.Name + "." + h.ClientConfig.Service.Namespace + ".svc"
 	u, err := a.serviceResolver.ResolveEndpoint(h.ClientConfig.Service.Namespace, h.ClientConfig.Service.Name)
```

Comment:
don't need to call `a.serviceResolver.ResolveEndpoint` here, right?

Reply:
if you're setting a custom dialer, you can set `cfg.Host = serverName` and just do the resolution in the dialer

Reply:
I just need the u.Path  here.

Reply:
However, currently all serviceResolver used, do not set `Path `,  so I can remove it here.